### PR TITLE
Split configuration options out of `uv-types`

### PR DESCRIPTION
**Pull request by zanieb**
Needed to prevent circular dependencies in my toolchain work (#2931). I think this is probably a reasonable change as we move towards persistent configuration too?

Unfortunately `BuildIsolation` needs to be in `uv-types` to avoid circular dependencies still. We might be able to resolve that in the future.

### CODE REVIEW

**Comment by zanieb** [comment]
Hm my problem is actually rooted in `uv_interpreter` being required by `uv_config` for `BuildIsolated::Shared`

**Review by charliermarsh — APPROVED** [review]
Some of the categorizations feel a bit off, e.g. should `requirements.rs` be in there?

File: Cargo.toml [hunk 1]
```diff
@@ -44,6 +44,7 @@ uv-normalize = { path = "crates/uv-normalize" }
 uv-requirements = { path = "crates/uv-requirements" }
 uv-resolver = { path = "crates/uv-resolver" }
 uv-types = { path = "crates/uv-types" }
+uv-config = { path = "crates/uv-config" }
```
[the remark this answers is not in the document]
Comment (zanieb) [reply Cargo.toml:47]: I'm down.

**Comment by charliermarsh** [comment]
Maybe leave it in types if you can. I might need to add some new types there related to requirements for hash-checking anyway.

**Pull request merged**